src/Execution/Arguments/ArgumentSet.php

```diff
@@ -100,13 +95,20 @@ public function enhanceBuilder($builder, array $scopes, Closure $directiveFilter
                 $value = $value->value;
             }
 
-            $argument
+            $filteredDirectives = $argument
                 ->directives
-                ->filter($directiveFilter)
-                ->each(function (ArgBuilderDirective $argBuilderDirective) use (&$builder, $value) {
-                    $builder = $argBuilderDirective->handleBuilder($builder, $value);
+                ->filter(function (Directive $directive): bool {
+                    return $directive instanceof ArgBuilderDirective;
                 });
 
+            if (!empty($directiveFilter)) {
+                $filteredDirectives = $filteredDirectives->filter($directiveFilter);
+            }
+
+            $filteredDirectives->each(function (ArgBuilderDirective $argBuilderDirective) use (&$builder, $value) {
+                $builder = $argBuilderDirective->handleBuilder($builder, $value);
+            });
+
```

Comment:
> I think we should apply the extra filter in addition to filtering for ArgBuilderDirective, because the function afterwards can only continue to work with such instances.

@spawnia do you mean like this?